packages/api/src/utils/client/httpClient.ts

```diff
@@ -265,7 +271,8 @@ export class HttpClient implements IHttpClient {
     try {
       const url = urlJoin(baseUrl, opts.url) + (opts.query ? "?" + stringifyQuery(opts.query) : "");
 
-      const headers = opts.headers || {};
+      const headers =
+        extraHeaders && opts.headers ? {...extraHeaders, ...opts.headers} : opts.headers || extraHeaders || {};
```

Comment:
doesn't the follow do achieve the same?

```suggestion
      const headers = {...extraHeaders, ...opts.headers};
```

as destructuring `undefined` and `null` will just be omitted

```node
> {...null}
{}
> {...undefined}
{}
```

apparently spread syntax on `null` is supposed to throw an error but can't reproduce that on node 18, but just in case could do

```suggestion
      const headers = {...(extraHeaders ?? {}), ...(opts.headers ?? {})};
```

Reply:
Functionally equivalent but your approach requires spreading all object's keys if only one headers option is specified